engine/src/main/java/org/hibernate/validator/internal/engine/constraintvalidation/ConstraintTree.java

```diff
@@ -208,12 +204,13 @@ private ValidationException getExceptionForNullValidator(Type validatedValueType
 	}
 
 	@SuppressWarnings("unchecked")
-	private ConstraintValidator<A, ?> getInitializedConstraintValidator(ConstraintValidatorManager constraintValidatorManager,
-			ConstraintValidatorFactory constraintValidatorFactory) {
-		ConstraintValidator<A, ?> validator = constraintValidatorManager.getInitializedValidator(
+	private ConstraintValidator<A, ?> getInitializedConstraintValidator(ValidationContext<?> validationContext) {
+		ConstraintValidator<A, ?> validator = validationContext.getConstraintValidatorManager().getInitializedValidator(
 				validatedValueType,
 				descriptor,
-				constraintValidatorFactory );
+				validationContext.getConstraintValidatorFactory(),
+				HibernateConstraintValidatorInitializationContextImpl.from( validationContext )
```

Comment:
I gave it some more thoughts and I think we should create the context at the `ValidatorImpl` level once and for all for now, and pass it to the `ValidationContext`. This would avoid creating a new object for each validation.

It might be necessary in the future but while it is not, I think it's worth avoiding the object creation.